R/mod_study_forms.R

```diff
@@ -193,6 +197,96 @@ mod_study_forms_server <- function(
       lapply(add_missing_columns(item_info, cols)[1, cols], isTRUE)
     })
     
+    observe({
+      reload_data(reload_data() + 1)
+      session$userData$update_checkboxes[[form]] <- NULL
+      session$userData$review_records[[form]] <- data.frame(id = integer(), reviewed = character())
+    }) |> 
+      bindEvent(r$subject_id, r$review_data,
+                ignoreInit = TRUE)
+    
+    observeEvent(session$userData$update_checkboxes[[form]], {
+      reload_data(reload_data() + 1)
+      checked <- session$userData$update_checkboxes[[form]]
+      
+      df <- table_data() |> 
+        dplyr::mutate(o_reviewed = dplyr::if_else(subject_id == r$subject_id, 
+                                                  lapply(o_reviewed, modifyList, list(updated = checked)),
+                                                  o_reviewed))
+      table_data(df)
+    })
+    
+    observeEvent(input$table_review_selection, {
+      # Update review values for session's user data
+      session$userData$update_checkboxes[[form]] <- NULL
+      session$userData$review_records[[form]] <-
```

Comment:
In essence it is still being used in that way because that's where the field `reviewed` is coming from. The new field `o_reviewed` is necessary for the datatable object. Each datatable is pivoted, so the datatable itself would contain multiple records, this needs to be consolidated in some way for a row level review to even be possible.